app/services/printer-service.js

```diff
@@ -0,0 +1,14 @@
+import Ember from "ember";
+
+export default Ember.Service.extend({
+  store: Ember.inject.service(),
+  allAvailablePrinters() {
+    let availablePrinters = this.get("store").peekAll("printer");
+    let printers = [];
+    availablePrinters.map(printer => {
+      let tag = printer.get("name");
+      printers.push({ name: tag, id: printer.get("id") });
+    });
```

Comment:
```js
    availablePrinters.map(printer => {
      printers.push({ 
         name: printer.get("name"), 
         id:   printer.get("id") 
      });
    });
```

Reply:
if it's using `map`, then there's no need to push anything into an array, you can just return it : )

```javascript
  return availablePrinters.map(printer => printer.getProperties('name', 'id'))
```